### ISSUE TITLE
Add support for Fabric

### ISSUE THREAD

**Issue by XieXiLin158** [SpigotMC/BungeeCord#3247]
Like title

**Comment by dayo05** [comment]
This is plugin, not mod. You don't need to install mod loader like forge, fabric, etc.

**Comment by Janmm14** [comment]
@XieYiLin158
Support for mods is not that easy.

The mod system would need the option to change the active mod synchronisation in the middle of the connection.

When new entities are added before version 1.16.5 or so, stuff is getting quite impossible because of needed entity id rewriting.

In the past years, there was no intention by md_5 to look into mod support for forge, so with fabric it should be similar.

What is needed for an implementarion is a documentation of the protocol changes the mod framework is doing.
And how mods are checked on client&server.
And a possibility is needed to redo that check (at least during a world swap, as bungee uses that to change servers)

**Comment by XieXiLin158** [comment]
OK,thanks.

**Comment by xIsm4** [comment]
> @XieYiLin158 Support for mods is not that easy.
> 
> The mod system would need the option to change the active mod synchronisation in the middle of the connection.
> 
> When new entities are added before version 1.16.5 or so, stuff is getting quite impossible because of needed entity id rewriting.
> 
> In the past years, there was no intention by md_5 to look into mod support for forge, so with fabric it should be similar.
> 
> What is needed for an implementarion is a documentation of the protocol changes the mod framework is doing. And how mods are checked on client&server. And a possibility is needed to redo that check (at least during a world swap, as bungee uses that to change servers)

Ligthfall does exactly that, they add a Custom handshake + a Upstreambridge for Forge and with the mod they give access.

**Comment by Janmm14** [comment]
> > @XieYiLin158 Support for mods is not that easy.
> > The mod system would need the option to change the active mod synchronisation in the middle of the connection.
> > When new entities are added before version 1.16.5 or so, stuff is getting quite impossible because of needed entity id rewriting.
> > In the past years, there was no intention by md_5 to look into mod support for forge, so with fabric it should be similar.
> > What is needed for an implementarion is a documentation of the protocol changes the mod framework is doing. And how mods are checked on client&server. And a possibility is needed to redo that check (at least during a world swap, as bungee uses that to change servers)
> 
> Ligthfall does exactly that, they add a Custom handshake + a Upstreambridge for Forge and with the mod they give access.

Well, we here cannot just go to said fork and copy their code into Bungee. If one wants to see such support updated in bungee, they could ask the creators if they'd be willing to allow usage in Bungee or even if they maybe create a PR into bungee so more people can benefit from it.

Before just now, I never knew such a fork even exists - there are far too many forks of bungee around.

Yes, it is more time consuming to wait for md_5's review and maybe changing things again, but as my PRs show, good ones do get accepted.

**Comment by Janmm14** [comment]
> > > @XieYiLin158 Support for mods is not that easy.
> > > The mod system would need the option to change the active mod synchronisation in the middle of the connection.
> > > When new entities are added before version 1.16.5 or so, stuff is getting quite impossible because of needed entity id rewriting.
> > > In the past years, there was no intention by md_5 to look into mod support for forge, so with fabric it should be similar.
> > > What is needed for an implementarion is a documentation of the protocol changes the mod framework is doing. And how mods are checked on client&server. And a possibility is needed to redo that check (at least during a world swap, as bungee uses that to change servers)
> > 
> > 
> > Ligthfall does exactly that, they add a Custom handshake + a Upstreambridge for Forge and with the mod they give access.
> 
> Well, we here cannot just go to said fork and copy their code into Bungee. If one wants to see such support updated in bungee, they could ask the creators if they'd be willing to allow usage in Bungee or even if they maybe create a PR into bungee so more people can benefit from it.
> 
> Before just now, I never knew such a fork even exists - there are far too many forks of bungee around.
> 
> Yes, it is more time consuming to wait for md_5's review and maybe changing things again, but as my PRs show, good ones do get accepted.

Uh and as a last point - @md-5 should really close these issues and PRs at some point,  maybe with some labels like

- no answer
- outdated
- too experimental
- out of scope

Unfortunaly, some PRs are also waiting on md-5's thoughts for months or even longer. :(
